xmtp_mls/src/utils/id.rs

```diff
@@ -17,3 +16,18 @@ pub fn get_message_id(
     id_vec.extend_from_slice(decrypted_message_bytes);
     sha256(&id_vec)
 }
+
+/// Relies on a client-created idempotency_key (which could be a timestamp)
+pub fn calculate_message_id(
+    group_id: &[u8],
+    decrypted_message_bytes: &[u8],
+    sender_account_address: &str,
+    idempotency_key: &str,
+) -> Vec<u8> {
+    let mut id_vec = Vec::new();
+    id_vec.extend_from_slice(group_id);
+    id_vec.extend_from_slice(decrypted_message_bytes);
+    id_vec.extend_from_slice(sender_account_address.as_bytes());
+    id_vec.extend_from_slice(idempotency_key.as_bytes());
```

Comment:
`decrypted_message_bytes` and `idempotency_key` are not a fixed length, so I wonder if collisions are possible by manipulating the message and idempotency_key.  The old `get_message_id()` gets around this problem by putting `decrypted_message_bytes` at the end. Maybe we can also put `decrypted_message_bytes` at the end, and do a `sha256` on the `idempotency_key` so it's always the same length?

Reply:
I put the message bytes as the last extended piece.
Clarifying ?:   are you suggesting:
`sha256(group_id, sender_account_address, sha256(idempotency_key), decrypted_message_bytes)`

Reply:
Yep! Curious what @neekolas thinks

Reply:
Still not sure how manipulating the idempotency key or message becomes an attack vector tbh. The sender account address and group id are part of the hash, so you can't replay anyone else's messages and get the same ID. And in the original the two are separated by the sender account address which can't be manipulated, so they won't bleed into one another.

Given this is all getting hashed at the end, we could also just put in separators between the fields if we thought this was a risk. Should be more performant than doing an extra `sha256`.

Reply:
Separators make sense, but we'd have to make sure the character set for the separator cannot be present in any of the fields. To explain more, imagine sender A has address abcdefg, and sender b has address bcdefgh. Sender A sends "hello", which will create an `id_vec` that contains "<group_id>helloabcdefg<idempotency_key>". Sender B can deliberately cause a collision by sending "helloa", and massage their idempotency key to match sender A.

Obviously difficult to do in practice, but if a small tweak makes edge cases like this impossible, I think it's worth making the security easier to reason about.

Reply:
Because we have the sender address in the middle, anything that isn’t a hex character should be a valid separator.